In [3]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

# Use the kagglehub client library to attach Kaggle resources like competitions, datasets, and models to your session
# Learn more about kagglehub: https://github.com/Kaggle/kagglehub/blob/main/README.md

import kagglehub
# kagglehub.dataset_download('<owner>/<dataset-slug>')

/kaggle/input/datasets/farehyasmine/datasetfordataaalyticss/Dataset for Data Analytics.xlsx


# import pandas 

In [4]:
import pandas as pd
print('done')

done


# LOAD DATASET

In [5]:
df=pd.read_excel('/kaggle/input/datasets/farehyasmine/datasetfordataaalyticss/Dataset for Data Analytics.xlsx')

In [6]:
df.head(5)

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

#  vérifier les valeurs manquante

In [8]:
df.isna().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

# la colonne CouponCode contient des valeurs manquantes =309


# les doublons

In [9]:
print(df.duplicated().sum())

0


# **all OrderIDSs are unique** 


In [10]:
duplicate_ids = df[df["OrderID"].duplicated(keep=False)]

print("Duplicate OrderIDs:", len(duplicate_ids))

duplicate_ids

Duplicate OrderIDs: 0


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice


In [11]:
#SPEAK ONE LANGUAGE 
df["Date"] = pd.to_datetime(df["Date"], errors="coerce")

In [12]:
print(df["Date"])

0      2023-01-04
1      2024-08-23
2      2024-02-27
3      2023-10-15
4      2025-05-08
          ...    
1195   2024-06-20
1196   2024-03-04
1197   2023-07-13
1198   2024-08-22
1199   2023-06-11
Name: Date, Length: 1200, dtype: datetime64[ns]


#  Les Valeurs fréquentes de CouponCode

In [13]:
df['CouponCode'].value_counts()

CouponCode
FREESHIP    313
WINTER15    292
SAVE10      286
Name: count, dtype: int64

In [14]:
#remplacer les valeurs manquantes par No Coupon
df['CouponCode']=df['CouponCode'].fillna('No Coupon')


In [15]:
# VERIFICATION APRE LE CHANGEMENT
df.isna().sum()

OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64

In [16]:
#verifier le totalprice 
df["CalculatedTotal"] = (
    df["Quantity"] * df["UnitPrice"]
).round(2)

df["TotalPrice"] = df["TotalPrice"].round(2)

errors = df[
    df["TotalPrice"] != df["CalculatedTotal"]
]

print("Incorrect TotalPrice:", len(errors))

Incorrect TotalPrice: 0


In [17]:
# supprimer la colon CalculatedTotal avant saved the dataset
df.drop(columns=["CalculatedTotal"], inplace=True)

In [18]:
df.to_excel(
    "Dataset_Cleaned.xlsx",
    index=False
)

print("Cleaned dataset saved successfully!")

Cleaned dataset saved successfully!
